internal/languages/python/analyzer/analyzer.go

```diff
@@ -120,10 +120,20 @@ func (analyzer *analyzer) analyzeAttribute(node *sitter.Node, visitChildren func
 }
 
 // foo["bar"]
+// foo[x]
+// globals()[y]
 func (analyzer *analyzer) analyzeSubscript(node *sitter.Node, visitChildren func() error) error {
 	objectNode := node.ChildByFieldName("value")
-	analyzer.builder.Dataflow(node, objectNode)
-	analyzer.lookupVariable(objectNode)
+	if objectNode.Type() == "identifier" {
```

Comment:
I think we want to do this unconditionally, `lookupVariable` will check the type anyway, and we'll want the dataflow association even if it's not a variable/identifier